apps/site/assets/ts/schedule/components/Map.tsx

```diff
@@ -109,20 +73,20 @@ interface IdHash {
 const shouldRemoveMarker = (id: string | null, idHash: IdHash): boolean =>
   id !== null && idHash[id] === true;
 
-interface State {
+interface EventData {
+  marker: Marker;
+}
+
+type DataFromSocketType = SocketEvent<EventData[]>;
+interface DataReducerState {
   channel: string;
   markers: Marker[];
 }
+type DataReducerAction = DataFromSocketType & { channel: string };
+type DataReducerType = Reducer<DataReducerState, DataReducerAction>;
 
-export const reducer = (
-  state: State,
-  actionWithChannel: ActionWithChannel
-): State => {
-  const { action, channel } = actionWithChannel;
-  if (channel !== state.channel && action.event !== "setChannel") return state;
```

Comment:
Should this conditional have been removed (or at least the first half now that `setChannel` is not a handled event anymore)? I'm curious if this is the reason for the failing test.

Reply:
Good callout. After looking into this a lot today, I am concluding that the channel comparison is no longer needed -- with the logic handled in the `useChannel` hook, it seems the reducer will only be working with data from the specified channel anyway. 

I'm not really sure where the intermixing of data from different channels was experienced or expected, but I couldn't find any problem on the Green Line page (which shows trains from the four distinct routes), and switching directions works as expected (the map doesn't show trains going in both directions, just the selected one.)

Have added an extra commit to clean up and adjust the failing test that's no longer needed.